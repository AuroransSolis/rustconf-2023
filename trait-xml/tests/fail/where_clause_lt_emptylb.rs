trait_xml::trait_xml! {
    <trait>
        <name>LifetimeClauseTest</name>
        <bounds>
            <lifetime>
                <name>'foo</name>
            </lifetime>
            <lifetime>
                <name>'bar</name>
            </lifetime>
            <lifetime>
                <name>'baz</name>
            </lifetime>
        </bounds>
        <where>
            <lifetime-clause>
                <lifetime>'foo</lifetime>
                <lifetime-bound></lifetime-bound>
            </lifetime-clause>
        </where>
    </trait>
}

fn main() {}
