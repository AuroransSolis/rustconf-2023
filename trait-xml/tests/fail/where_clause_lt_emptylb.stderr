error: error parsing lifetime parameter: expected lifetime, found `<`. caller: `::trait_xml::trait_xml_parse_lifetime_clause`
  --> tests/fail/where_clause_lt_emptylb.rs:1:1
   |
1  | / trait_xml::trait_xml! {
2  | |     <trait>
3  | |         <name>LifetimeClauseTest</name>
4  | |         <bounds>
...  |
21 | |     </trait>
22 | | }
   | |_^
   |
   = note: this error originates in the macro `$crate::trait_xml_parse_lifetime_bound` which comes from the expansion of the macro `trait_xml::trait_xml` (in Nightly builds, run with -Z macro-backtrace for more info)
